src/mito2/src/compaction.rs

```diff
@@ -444,11 +445,15 @@ async fn find_ttl(
         return Ok(Some(table_ttl));
     }
 
-    let ttl = schema_metadata_manager
-        .get_schema_options_by_table_id(table_id)
-        .await
-        .context(GetSchemaMetadataSnafu)?
-        .and_then(|options| options.ttl);
+    let ttl = tokio::time::timeout(
+        crate::config::FETCH_TTL_TIMEOUT,
+        schema_metadata_manager.get_schema_options_by_table_id(table_id),
```

Comment:
We use this method to fetch options so `FETCH_OPTION_TIMEOUT` is more proper.
```suggestion
        crate::config::FETCH_OPTION_TIMEOUT,
        schema_metadata_manager.get_schema_options_by_table_id(table_id),
```